datacube_stats/main.py

```diff
@@ -82,6 +82,18 @@ def list_statistics(ctx, param, value):
     ctx.exit()
 
 
+def gather_tile_indexes(tile_index, tile_index_file):
+    if not tile_index and not tile_index_file:
+        return None
+
+    if tile_index:
+        return [tile_index]
```

Comment:
Probably worth to add `assert` to indicate that you expect only one of the inputs to be non-emty. OR if you want support both properly: read from file and add one more from a command line.